src/sigchain/sigchain.iced

```diff
@@ -8,7 +8,10 @@ ie = require('iced-error')
 UID_LEN = 32
 exports.SIG_ID_SUFFIX = SIG_ID_SUFFIX = "0f"
 
-
+strip_final_newline = (buf) ->
+  s = buf.toString('utf8')
```

Comment:
If you want to you can check `buf[-1...].equals Buffer "\n"` without doing the string conversion. I don't have a strong preference either way.